File: Java/DesignPatterns/src/main/java/designPatterns/behavioral/command/ConcreteCommand.java
package designPatterns.behavioral.command;

public class ConcreteCommand {
  static class A implements Command {
    private Receiver receiver;

    public A(Receiver receiver) {
      this.receiver = receiver;
    }

    @Override
    public String execute() {
      return receiver.executeA();
    }
  }

  static class B implements Command {
    private Receiver receiver;

    public B(Receiver receiver) {
      this.receiver = receiver;
    }

    @Override
    public String execute() {
      return receiver.executeB();
    }
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/behavioral/command/ConcreteCommand.kt
package behavioral.command

sealed class ConcreteCommand {
  data class A(private var receiver: Receiver) : Command {
    override fun execute(): String {
      return receiver.executeA()
    }
  }

  data class B(private var receiver: Receiver) : Command {
    override fun execute(): String {
      return receiver.executeB()
    }
  }
}
